internal/services/jsonrpc/usersvc/service.go

```diff
@@ -494,9 +499,36 @@ func (svc *Service) Call(ctx context.Context, req *userjson.CallRequest) (*userj
 	ctxExec, cancel := context.WithTimeout(ctx, svc.readTxTimeout)
 	defer cancel()
 
-	readTx := svc.db.BeginDelayedReadTx()
+	// we use a basic read tx since we are subscribing to notices,
+	// and it is therefore pointless to use a delayed tx
+	readTx, err := svc.db.BeginReadTx(ctx)
+	if err != nil {
+		return nil, jsonrpc.NewError(jsonrpc.ErrorNodeInternal, "failed to start read tx", nil)
+	}
 	defer readTx.Rollback(ctx)
 
+	logCh := make(chan string)
+	var logs []string
+
+	done, err := readTx.Subscribe(ctx, logCh)
+	if err != nil {
+		close(logCh)
+		return nil, jsonrpc.NewError(jsonrpc.ErrorNodeInternal, "failed to subscribe to notices", nil)
+	}
+	defer done()
+	defer close(logCh) // close logCh after done to avoid potential infinite blocking.
```

Comment:
As a general rule, the sender closes, but it's ok if the pattern is clear and there's no possible send on the channel past this point.

I'm a little uncertain about the infinite blocking comment.  Will dig in briefly.